### DISCUSSION TITLE
Ignore Build Step (and more) broken

### DISCUSSION THREAD

**Discussion by MaitreyaBuddha** [vercel/community#6248] [Bug Report]
### Page to Investigate

https://vercel.com/docs/monorepos#ignoring-the-build-step

### Steps to Reproduce

Ignore build still still builds my project with any ignore build step command that returns `0`.

# Given
![image](https://github.com/vercel/community/assets/723393/ddf765b9-92a7-4b6b-a1e0-25b452785ffc)
![image](https://github.com/vercel/community/assets/723393/674a81bc-7f3a-47da-a63e-b78609360cf3)
Code:
```
const NEXT_PUBLIC_APP_ENV = process.env.NEXT_PUBLIC_APP_ENV 
console.log(`Using environment ${NEXT_PUBLIC_APP_ENV}`)
```

# When
`git push`
`   a5d0cf5..b6ec95d  kelly/buildtest -> kelly/buildtest`

# Expected
Build does not run.

# Actual
Build is created.
![image](https://github.com/vercel/community/assets/723393/80151b63-f68e-4c8f-a1b8-043fe15d4f45)

# Also
I wouldn't have noticed this issue if build caching was project-specific. The underlying issue is that my environment variables from any previously built project are being brought in with the cache. It's not deterministic which builds first.

TL:DR:
`Restored build cache`
`build: Using environment prod`
```
Restored build cache
Running "vercel build"
Vercel CLI 33.6.1
Detected `pnpm-lock.yaml` version 6 generated by pnpm 8
Running "install" command: `pnpm install`...
Lockfile is up to date, resolution step is skipped
Already up to date
Done in 2.1s
Detected Next.js version: 13.4.19
Running "pnpm turbo build"
• Running build
• Remote caching enabled
build: cache hit, replaying output cae2575c5f0f6575
build: 
build: > @ build /vercel/path0
build: > next build
build: 
build: Using environment prod
```
